src/server/game/Entities/Object/Object.h

```diff
@@ -396,7 +397,10 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
         void SummonCreatureGroup(uint8 group, std::list<TempSummon*>* list = nullptr);
 
         Creature*   FindNearestCreature(uint32 entry, float range, bool alive = true) const;
+        std::list<Creature*> FindAllCreaturesByEntry(uint32 entry, float range, bool alive = true) const;
```

Comment:
Not sure what the point would be. Lazy initialization? In 99% of the situations needed for scripts there is a solution that doesn't require grid lookups

Reply:
@Warpten , so the idea was that this boss spawns multiple creatures of one entry using a spell/gob so in order targeting to work properly after spawned I had to have a way of getting all unique creatures of entry and then script them accordingly. I have to admit this is my first script so additional feedback/guidance is desired, also there might be existing way of achieving what I did that I missed, but in general this was the idea.

Reply:
i think grid lookups arent as bad as they used to be but generally speaking all you need is to hook JustSummoned in the summoner and store a summon list

f the boss isn't the one summoning the unit (either directly or via a spell), script the summoned unit with a IsSummonedBy hook, retrieve a pointer to the boss from instance script, and call BossAI::JustSummoned(me).

Example:

https://github.com/TrinityCore/TrinityCore/blob/master/src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp#L588

https://github.com/TrinityCore/TrinityCore/blob/master/src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp#L1022

Reply:
@Warpten, thanks for the suggestions, I remember trying JustSummoned hook but as you said its not the boss who summons the spiders its the effect of game object, anyway, I think I am already keeping a summon list and despawning afterwards anyway, what is this all about? :) This method I am using in order to get all unique creatures nearby by given entry and order them to start attack one by one.

Reply:
> script the summoned unit with a IsSummonedBy hook, retrieve a pointer to the boss from instance script, and call BossAI::JustSummoned(me).

what happened to this ? this will not require any change to "game" project

Reply:
Well, it turns out that Spider Egg is summoning the spiders (it all starts from [here](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/SpellEffects.cpp#L2231) which leads to [here](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Object/Object.cpp#L1972) where `ToUnit()` call fails because Spider Egg is gameobject not a unit ultimately failing to set summoner for this tempsummon (the spider) then this check over [here](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Creature/TemporarySummon.cpp#L207) fails and does not raise the aforementioned hooks.

@Warpten , @jackpoz 

Question is: Is this adequate behavior of the core or we are missing functionality? (To me it sounds that it is possible for gameobject to be summoner but I am new in the project)

Reply:
Yes game object could be a summoner, but there's an issue, https://github.com/TrinityCore/TrinityCore/issues/23131

Reply:
Yes, I assume we need logic in the core to handle gameobject summoners so they can properly raise further hooks afterwards.

Reply:
better fix the gameobject summoning in another PR first then

Reply:
@jackpoz I think it will require a lot of core changes and I don't think I will be the best person to do it since I am new to this project but I will try to open one issue explaining whats expected and what is missing imo.

Reply:
here it is https://github.com/TrinityCore/TrinityCore/issues/23242